### PR TITLE
Add zero confirmations option, #49

### PR DESCRIPTION
**Pull request by DarrenWestwood**
Accept payments on 0, 1, 2 bitcoin confirmations
Issue #49

### CODE REVIEW

**Comment by DarrenWestwood** [comment]
`checkCbTransID($txid)` verifies the same transaction is not added to the invoice multiple times.

**Comment by blockonomics** [comment]
[This logic](https://github.com/blockonomics/whmcs-bitcoin-plugin/blob/84c77537c05ddf2e49695e5fc86deebb8e11c61b/payment.php#L123) is comparing with status code directly and will break. Please address

**Comment by DarrenWestwood** [comment]
> [This logic](https://github.com/blockonomics/whmcs-bitcoin-plugin/blob/84c77537c05ddf2e49695e5fc86deebb8e11c61b/payment.php#L123) is comparing with status code directly and will break. Please address

Logic looks okay as the code only runs if order is not completed / has not received the total yet. Tested the functionality of multiple payments for individual order is working correctly.

**Comment by blockonomics** [comment]
I also found that [this code](https://github.com/blockonomics/whmcs-bitcoin-plugin/blob/master/payment.php#L144) is executing incorrectly when _status==0 || status==1_ . We need to alteast check following test cases:

1. Zero / one confirmation - invoice paid in one go
2. Zero / one confirmation - invoice underpaid
3. Zero / one confirmation - invoice underpaid with slack 
4. Zero / one confirmation - invoice paid with multiple tx
5. Zero / one confirmation - Try paying with existing payment in progress

**Comment by DarrenWestwood** [comment]
> [This logic](https://github.com/blockonomics/whmcs-bitcoin-plugin/blob/84c77537c05ddf2e49695e5fc86deebb8e11c61b/payment.php#L123) is comparing with status code directly and will break. Please address

756ecec addresses this

> I also found that [this code](https://github.com/blockonomics/whmcs-bitcoin-plugin/blob/master/payment.php#L144) is executing incorrectly when _status==0 || status==1_

07a27b1 only adjust btc_amount in database if tx has not been received or tx is completed/underpaid

**Comment by DarrenWestwood** [comment]
1) invoice paid in one go - Working (Directs user once specified confirmations reached) 
2) invoice underpaid - Working (Tx added to invoice on specified confirmations)
3) invoice underpaid with slack - Working (Slack working with paid in one go and multiple tx)
4) invoice paid with multiple tx - Working (Tx correctly added to invoice on specified confirmations)
5) existing payment in progress - Working (If transaction received but less than the specified confirmations, _Payment is pending_ message is displayed)

File: payment.php [hunk 1]
```diff
@@ -125,23 +125,27 @@ function generate_address($blockonomics, $ca) {
...
+		} elseif ($existing_order['status'] < $confirmations) {
 			$ca->assign('pending', true);
 			$ca->assign('txid', $existing_order['txid']);
-		} elseif ($existing_order['status'] == 2) {
+		} elseif ($existing_order['status'] >= $confirmations) {
 			$btc_address = generate_address($blockonomics, $ca);
 			$blockonomics->updateOrderAddress($order_id, $btc_address);
+			$btc_amount = $blockonomics->getBitcoinAmount($fiat_amount, $currency);
+			$ca->assign('btc_amount', $btc_amount / 1.0e8);
+			$blockonomics->updateOrderExpected($order_id, $btc_amount, $fiat_amount);
```
Comment (blockonomics) [inline payment.php]: Is it not necessary to existing order status to *-1* again?

File: modules/gateways/callback/blockonomics.php [hunk 2]
```diff
@@ -45,7 +45,13 @@
 $invoiceId = $order['order_id'];
 $bits = $order['bits'];
 
-if($status == 0) {
+if($status != 2 && $status != 1 && $status != 0) {
```
Comment (blockonomics) [inline modules/gateways/callback/blockonomics.php]: Good check, but not necessary can be removed

File: payment.php (hunk 1 above)
Comment (jusasiiv) [reply payment.php]: It is done inside the function https://github.com/blockonomics/whmcs-bitcoin-plugin/blob/324c3433c7bdce74da6539e48996ca5f4894b6e9/modules/gateways/Blockonomics/Blockonomics.php#L462

File: modules/gateways/callback/blockonomics.php (hunk 2 above)
Comment (DarrenWestwood) [reply modules/gateways/callback/blockonomics.php]: Removed with commit 1232446

**Pull request merged**